src/peer-store/README.md

```diff
@@ -1,3 +1,122 @@
-# Peerstore
+# PeerStore
 
-WIP
\ No newline at end of file
+Libp2p's PeerStore is responsible for keeping an updated register with the relevant information of the known peers. It should be the single source of truth for all peer data, where a subsystem can learn about peers' data and where someone can listen for updates. The PeerStore comprises four main components: `addressBook`, `keyBook`, `protocolBook` and `metadataBook`.
+
+The PeerStore manages the high level operations on its inner books. Moreover, the PeerStore should be responsible for notifying interested parties of relevant events, through its Event Emitter.
+
+## Data gathering
```

Comment:
I think we can do it in a followup PR but I think we should simplify the data gathering and consumption sections. Reading through this it was hard to figure out where data comes from and who/what/why it's being consumed. I think it would be good to be very direct about what should actually be putting data into the peer store so it's clear, and how someone can get data out of the peer store.

IE:

## Submitting records to the PeerStore

### Identify
- The Identify protocol automatically runs on every connection when multiplexing is enabled. The protocol will put the multiaddrs and protocols provided by the peer to the PeerStore. As these records are directly from the peer, they should be considered the source of truth and weighted accordingly.

...

## Retrieving records from the PeerStore

### Protocols
- *how to get existing protocols*
- *how to listen for protocol changes*

...

Reply:
Yeah, I totally agree to use that text format. I will PR with that on a followup